db/helpers/generatorUtil.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/helpers/generatorUtil.js
+ */
+const common = require('./common');
+const dbErrors = require('../dbErrors');
+
+/**
+ * Reject the request if collectorNames contain duplicate names
+ * @param {Array} collectorNames Array of strings
+ * @returns {Promise} empty if validation passed, reject otherwise
+ */
+function validateCollectorNames(collectorNames) {
+  if (common.checkDuplicatesInStringArray(collectorNames)) {
+    const err = new dbErrors.DuplicateCollectorError();
+    err.resourceType = 'Collector';
+    err.resourceKey = collectorNames;
+    return Promise.reject(err);
+  }
+
+  return Promise.resolve();
+}
+
+/**
+ * If collectors exist, return a Promise with an
+ * Array of collector objects referenced by collectorNames.
+ * If collector names are not supplied, return a Promise
+ * with an empty array
+ * If collector names are invalid, reject with error.
+
+ * @param {Object} seq The sequelize object
+ * @param {Array} collectorNames Array of Strings
+ * @param {Function} whereClauseForNameInArr Passed in from API
+ * @returns {Promise} with an array if check passed, error otherwise
+ */
+function checkCollectorsExist(seq,
+  collectorNames, whereClauseForNameInArr) {
+  if (!collectorNames || !collectorNames.length) {
+    return [];
+  }
+
+  const options = {};
+  options.where = whereClauseForNameInArr(collectorNames || []);
```

Comment:
how can it ever get to `|| []` since you returned on line 46 if collectorNames is falsey or empty array?